lib/rules/jsx-pascal-case.js

```diff
@@ -14,7 +14,7 @@ const jsxUtil = require('../util/jsx');
 // ------------------------------------------------------------------------------
 
 const PASCAL_CASE_REGEX = /^([A-Z0-9]|[A-Z0-9]+[a-z0-9]+(?:[A-Z0-9]+[a-z0-9]*)*)$/;
-const ALL_CAPS_TAG_REGEX = /^[A-Z0-9]+$/;
+const ALL_CAPS_TAG_REGEX = /^[A-Z0-9_]+$/;
```

Comment:
i think we should ensure that it's `A_Z` or `A` but not `A_` or `_A` or `__` or `A__B`, with tests.

Reply:
Great catch, thanks! I added that. Don't really want to get into RegEx voodoo but this one seems simple enough: https://regex101.com/r/kd5Fma/1/